### PR TITLE
Add `@setCache`, `@deleteCache` and `@cleanInput` directives

### LINKED ISSUE #1772: Invalidate caches created by `@cache`
Yes I want to help you develop a lighthouse.

**What problem does this feature proposal attempt to solve?**

To delete the cache

```graphql
extend type Query {
    categories: [Category!]! @all @cache(key: "categories") @softDeletes
    categories2: [Category!]! @all @cache @softDeletes # If the key is not filled, use categories2 itself by default categories2
}

extend type Mutation {
    createCategory(input: CategoryInput! @spread): Category! @trim @create @cache(key: "categoryAll", delete: true)
}
```

**Which possible solutions should be considered?**

When creating, editing and deleting (operations performed on the database), by adding a key and deleting it to @cache, we can delete it using cache. And save with the desired name using the key

### PR DESCRIPTION
**Pull request by Rasool-deldar**
- [x] Added or updated tests
- [x] Documented user facing changes
- [ ] Updated CHANGELOG.md

Resolves #1772

**Changes**

Several features were added to the package
1 - @setCache
2 - @deleteCache
3 - @cleanInput

**Breaking changes**

None.

### CODE REVIEW

**Comment by spawnia** [comment]
I don't see how those features are supposed to be used and cannot be sure they are working. In order for me to consider inclusion, they need to be tested.

**Comment by stayallive** [comment]
For reference, this is related to #1772.

File: src/Schema/Directives/CleanInputDirective.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    protected function transformArgumentSet(ArgumentSet $argumentSet): ArgumentSet
+    {
+        foreach ($argumentSet->arguments as $argument) {
+            $argument->value = $this->sanitize($argument->value);
+        }
+
+        return $argumentSet;
+    }
+
+    protected function transformLeaf($value)
```
Comment (MarkG-IEL) [inline src/Schema/Directives/CleanInputDirective.php:76]: This set feels arbitrary. Wouldn't built-in PHP functions be better for this?

**Comment by spawnia** [comment]
I don't think those features are well thought out or coherent. I am open to smaller pull requests with more polish.

**Pull request closed without merging**